Source/WebCore/page/Quirks.cpp

```diff
@@ -2005,6 +2005,21 @@ bool Quirks::needsMozillaFileTypeForDataTransfer() const
     return false;
 }
 
+// bing.com rdar://126573838
+bool Quirks::needsBingGestureEventQuirk(EventTarget* target) const
+{
+    if (!needsQuirks())
+        return false;
+
+    auto url = topDocumentURL();
+    if (url.host() == "www.bing.com"_s && startsWithLettersIgnoringASCIICase(url.path(), "/maps"_s)) {
+        if (RefPtr element = dynamicDowncast<Element>(target))
+            return element->classList().contains("atlas-map-canvas"_s);
```

Comment:
Maybe `hasClassName` instead? That way it won't force unnecessary creation of `ElementRareData`.